pkg/apis/pipeline/v1beta1/pipelinerun_conversion.go

```diff
@@ -208,6 +215,160 @@ func (ptrs *PipelineTaskRunSpec) convertFrom(ctx context.Context, source v1.Pipe
 	ptrs.ComputeResources = source.ComputeResources
 }
 
+// ConvertTo implements apis.Convertible
+func (prs *PipelineRunStatus) ConvertTo(ctx context.Context, sink *v1.PipelineRunStatus) error {
+	sink.Status = prs.Status
+	sink.StartTime = prs.StartTime
+	sink.CompletionTime = prs.CompletionTime
+	sink.Results = nil
+	for _, pr := range prs.PipelineResults {
+		new := v1.PipelineRunResult{}
+		pr.convertTo(ctx, &new)
+		sink.Results = append(sink.Results, new)
+	}
+	if prs.PipelineSpec != nil {
+		sink.PipelineSpec = &v1.PipelineSpec{}
+		err := prs.PipelineSpec.ConvertTo(ctx, sink.PipelineSpec)
+		if err != nil {
+			return err
+		}
+	}
+	sink.SkippedTasks = nil
+	for _, st := range prs.SkippedTasks {
+		new := v1.SkippedTask{}
+		st.convertTo(ctx, &new)
+		sink.SkippedTasks = append(sink.SkippedTasks, new)
+	}
+	sink.ChildReferences = nil
+	for _, cr := range prs.ChildReferences {
+		new := v1.ChildStatusReference{}
+		cr.convertTo(ctx, &new)
+		sink.ChildReferences = append(sink.ChildReferences, new)
+	}
+	sink.FinallyStartTime = prs.FinallyStartTime
+	if prs.Provenance != nil {
+		new := v1.Provenance{}
+		prs.Provenance.convertTo(ctx, &new)
+		sink.Provenance = &new
+	}
+	// If embedded-status is set to "both", both ChildReferences and TaskRuns/Runs
+	// will be populated. In this case, use the value from ChildReferences.
+	// TODO: to handle conversion when `embedded-status` feature flag is set to full or both as in #5813
+	if sink.ChildReferences == nil {
+		if prs.TaskRuns != nil {
+			sink.ChildReferences = append(sink.ChildReferences, convertTaskRunsToChildReference(ctx, prs.TaskRuns)...)
+		}
+		if prs.Runs != nil {
+			sink.ChildReferences = append(sink.ChildReferences, convertRunsToChildReference(ctx, prs.Runs)...)
+		}
+	}
+	return nil
+}
+
+// ConvertFrom implements apis.Convertible
+func (prs *PipelineRunStatus) ConvertFrom(ctx context.Context, source *v1.PipelineRunStatus) error {
+	prs.Status = source.Status
+	prs.StartTime = source.StartTime
+	prs.CompletionTime = source.CompletionTime
+	prs.PipelineResults = nil
+	for _, pr := range source.Results {
+		new := PipelineRunResult{}
+		new.convertFrom(ctx, pr)
+		prs.PipelineResults = append(prs.PipelineResults, new)
+	}
+	if source.PipelineSpec != nil {
+		newPipelineSpec := PipelineSpec{}
+		err := newPipelineSpec.ConvertFrom(ctx, source.PipelineSpec)
+		if err != nil {
+			return err
+		}
+		prs.PipelineSpec = &newPipelineSpec
+	}
+	prs.SkippedTasks = nil
+	for _, st := range source.SkippedTasks {
+		new := SkippedTask{}
+		new.convertFrom(ctx, st)
+		prs.SkippedTasks = append(prs.SkippedTasks, new)
+	}
+	cfg := config.FromContextOrDefaults(ctx)
+	if cfg.FeatureFlags.EmbeddedStatus == config.FullEmbeddedStatus || cfg.FeatureFlags.EmbeddedStatus == config.BothEmbeddedStatus {
+		return fmt.Errorf("Invalid EmbeddedStatus value in v1: 'taskRuns' and 'runs' status are deprecated in v1, and cannot be populated by conversion from v1 childReferences")
```

Comment:
I'd maybe word this as `PipelineRun .status.taskRuns and .status.runs fields do not exist in v1, and cannot be populated by conversion from a v1 PipelineRun to a v1beta1 PipelineRun when the "embedded-status" feature flag is not set to "minimal" or "both"`.

Also might be worth finally remembering to put `embedded-status` in `config/config-feature-flags.yml`, which I for some reason never remembered to do, so that the flag shows up in `kubectl get configmap -n tekton-pipelines feature-flags -o yaml`, but that can be a separate PR.

Reply:
Thanks, updated the error message. And created https://github.com/tektoncd/pipeline/issues/5824 to track this down.